src/ts/components/subtitlesettingtoggle.ts

```diff
@@ -0,0 +1,130 @@
+import {ToggleButton, ToggleButtonConfig} from './togglebutton';
+import {Button, ButtonConfig} from './button';
+import {Container, ContainerConfig} from './container';
+import {Label, LabelConfig} from './label';
+import {SettingsPanel} from './settingspanel';
+import {SubtitleSettingsPanel} from './subtitlesettingspanel';
+import {UIInstanceManager} from '../uimanager';
+import {DOM} from '../dom';
+import {EventDispatcher, Event, NoArgs} from '../eventdispatcher';
+
+export interface SubtitleSettingsButtonConfig extends ButtonConfig {
+  subtitleSettingsPanel: SubtitleSettingsPanel;
+  settingsPanel: SettingsPanel;
+}
+
+/**
+ * A button that toggles the option menu for subtitles
+ */
+export class SubtitleSettingsOpener extends Button<ButtonConfig> {
```

Comment:
Please also move all extracted components from this file into the `subtitlesettings` subdirectory